core/src/scene/importer.cpp

```diff
@@ -0,0 +1,252 @@
+#include "importer.h"
+#include "platform.h"
+#include "scene/sceneLoader.h"
+#include "util/topologicalSort.h"
+#include "yaml-cpp/yaml.h"
+
+#include <regex>
+
+using YAML::Node;
+using YAML::NodeType;
+
+namespace Tangram {
+
+Node Importer::applySceneImports(const std::string& scenePath) {
+
+    if (loadScene(scenePath)) {
+        return importScenes(scenePath);
+    }
+
+    return Node();
+}
+
+std::string Importer::getFilename(const std::string& filePath) {
+    std::string sceneName = filePath;
+    std::regex r("[^//]+$");
+    std::smatch match;
+    if (std::regex_search(sceneName, match, r)) { sceneName = match[0]; }
+    return sceneName;
+}
+
+std::string Importer::normalizePath(const std::string &path,
+                                         const std::string &parentPath) {
+
+    //TODO: Handle network urls
+
+    if (path[0] == '/') {
+        //absolute path, return as it is
+        return path;
+    }
+
+    std::regex r("[^//]+$");
+    return std::regex_replace(parentPath, r, path);
+}
+
+void Importer::normalizeSceneImports(Node& root, const std::string& parentPath) {
+    if (Node import = root["import"]) {
+        if (import.IsScalar()) {
+            import = normalizePath(import.Scalar(), parentPath);
+        } else if (import.IsSequence()) {
+            for (Node imp : import) {
+                if (imp.IsScalar()) { imp = normalizePath(imp.Scalar(), parentPath); }
+            }
+        }
+    }
+}
+
+void Importer::setNormalizedTexture(Node& texture, const std::vector<std::string>& names,
+        const std::string& parentPath) {
+
+    for (size_t index = 0; index < names.size(); index++) {
+        auto& name = names[index];
+        if (m_textureNames.find(name) == m_textureNames.end()) {
```

Comment:
I don't think that we need that test. If we have two scenes that reference the same texture, this will fail to normalize the texture path where one imports the other, we will fail in normalizing the path and the second path will try to load a resource that is not relevant to its own resource root.
In general I think there's also an issue with the texture management within the scene, where they should be stored by URI at some point. I will create an issue and look into fixing that texture management.